### PR TITLE
Optimizes the search query so that it performs reasonably for large repos

### PR DESCRIPTION
**Pull request by dbernstein**
**Optimizes the search query so that it performs reasonably for large repos**
* * *

**JIRA Ticket**: https://fedora-repository.atlassian.net/browse/FCREPO-3733

* Other Relevant Links (Mailing list discussion, related pull requests, etc.)

# What does this Pull Request do?
This PR optimizes the search query when the user specifies the default fields or explicitly returns or filters on rdf_type.
# How should this be tested?
Run on a mysql backed  repo with at least a million items.

This should return in a few seconds:  
```
http://fcrepo-staging.eba-muvve2sq.us-east-1.elasticbeanstalk.com/fcrepo/rest/fcr:search
```

This should be lightning quick
```
http://fcrepo-staging.eba-muvve2sq.us-east-1.elasticbeanstalk.com/fcrepo/rest/fcr:search?max_results=10&condition=rdf_type=http://example.com/type
```

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers

### CODE REVIEW

File: fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/DbSearchIndexImpl.java [hunk 1]
```diff
@@ -840,7 +837,7 @@ private void cleanupTransaction(final String txId) {
     private List<String> toggleForeignKeyChecks(final boolean enable) {
 
         if (isPostgres()) {
-            return EMPTY_LIST;
+            return (List<String>)EMPTY_LIST;
```
Comment (whikloj) [inline fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/DbSearchIndexImpl.java:840]: You could just make this `return Collections.emptyList();`

File: fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraSearch.java [hunk 2]
```diff
@@ -93,7 +94,9 @@ public Response doSearch(@QueryParam(value = "condition") final List<String> con
                              @DefaultValue("100") @QueryParam("max_results") final int maxResults,
                              @DefaultValue("0") @QueryParam("offset") final int offset,
                              @DefaultValue("asc") @QueryParam("order") final String order,
-                             @QueryParam("order_by") final String orderBy) {
+                             @QueryParam("order_by") final String orderBy,
+                             @DefaultValue("false") @QueryParam("include_total_result_count")
```
[the remark this answers is not in the document]
Comment (whikloj) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraSearch.java:98]: Actually its the JavaDoc that is incorrect, this should be `false` by default

[the remark this answers is not in the document]
Comment (dbernstein) [reply fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraSearch.java:98]: I'll fix the javadoc

File: fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/DbSearchIndexImpl.java (hunk 1 above)
Comment (dbernstein) [reply fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/DbSearchIndexImpl.java:840]: will do,.

**Review by whikloj — APPROVED** [review]
👍

**Pull request merged**